unstract/workflow-execution/src/unstract/workflow_execution/execution_file_handler.py

```diff
@@ -145,3 +167,41 @@ def create_execution_dir_path(
         )
         execution_dir.mkdir(parents=True, exist_ok=True)
         return str(execution_dir)
+
+    @classmethod
+    def get_execution_dir(
+        cls, workflow_id: str, execution_id: str, organization_id: str
+    ) -> str:
+        """Create the directory path for storing execution-related files.
+
+        Parameters:
+        - workflow_id (str): Identifier for the workflow.
+        - execution_id (str): Identifier for the execution.
+        - organization_id (Optional[str]):
+            Identifier for the organization (default: None).
+
+        Returns:
+        str: The directory path for the execution.
+        """
+        execution_dir = f"/execution/{organization_id}/{workflow_id}/{execution_id}"
```

Comment:
@muhammad-ali-e I suggest building paths using `Pathlib`

Reply:
Updated to use Pathlib.

While FS APIs expect string paths, using Pathlib offers better handling for edge cases like missing or extra slashes, ensuring cleaner and more reliable path construction. If we need string make it string once path builded.